src/pairVmaf.ts

```diff
@@ -29,25 +29,39 @@ export async function pairVmafWithResolutionAndBitrate(
     }[]
   >();
   logger.info(`Loading VMAF data from ${directoryWithVmafFiles}...`);
-  const vmafs = await getVmaf(directoryWithVmafFiles, onProgress);
+  const analysisData = await getAnalysisData(
+    directoryWithVmafFiles,
+    onProgress
+  );
   let counter = 1;
-  const bitrates: Record<string, number> | undefined = probeBitrate
-    ? await getBitrates(
-        vmafs.map((vmaf) => path.resolve(directoryWithVmafFiles, vmaf.filename))
-      )
-    : undefined;
+  // Reinterpret bitrateList as a Record type
+  const bitrates: Record<string, number> | undefined =
```

Comment:
The logic in the ternary operator looks wrong. It seems even if analysisData.bitrateList exists it would still call getBitrates?

Reply:
Are you sure? As I see it getBitrates() function will not be run if analysisData.bitrateList exists and evaluates to a non-empty list.

Reply:
If the result of the reduce operation returns a non-empty object, the bitrates variable will be assigned the result of the reduce. If it is falsy, then await getBitrates(...) will be executed.

Reply:
gustav was right!